homeassistant/components/netgear/button.py

```diff
@@ -0,0 +1,85 @@
+"""Support for Netgear Button."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, KEY_COORDINATOR, KEY_ROUTER
+from .router import NetgearRouter, NetgearRouterEntity
+
+
+@dataclass
+class NetgearButtonEntityDescriptionRequired:
+    """Required attributes of NetgearButtonEntityDescription."""
+
+    action: Callable
```

Comment:
We can make this more specific

```suggestion
    action: Callable[[NetgearRouter], None]
```

Reply:
@frenck apperently this caused mypy errors, I am not that good in typing. Do you know how to fix this?